src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.FunctionalTest/tests/Hover2_2.test.ts

```diff
@@ -0,0 +1,158 @@
+        assert.equal(hoverResult!.length, 1, 'Should not have a hover result for InputTagHelper because it does not have the correct attrs yet.');
+
+        await editor.edit(edit => edit.insert(firstLine, '<input asp-for="D" class="someName" />\n'));
+        hoverResult = await vscode.commands.executeCommand<vscode.Hover[]>(
+            'vscode.executeHoverProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(0, 3));
+
+        assert.ok(hoverResult, 'Should have a hover result for InputTagHelper.');
+        if (hoverResult) {
+            assert.equal(hoverResult.length, 2, 'Something else may be providing hover results');
+
+            const envResult = hoverResult[0];
+            const expectedRange = new vscode.Range(
+                new vscode.Position(0, 3),
+                new vscode.Position(0, 3));
+            assert.deepEqual(envResult.range, expectedRange, 'TagHelper range should be <input>');
+            const mStr = envResult.contents[0] as vscode.MarkdownString;
+            assert.ok(mStr.value.includes('InputTagHelper'), `InputTagHelper not included in '${mStr.value}'`);
+        }
+
+        hoverResult = await vscode.commands.executeCommand<vscode.Hover[]>(
+            'vscode.executeHoverProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(0, 8));
+
+        assert.ok(hoverResult, 'Should have a hover result for asp-for');
+        if (hoverResult) {
+            assert.equal(hoverResult.length, 1, 'Something else may be providing hover results');
+
+            const envResult = hoverResult[0];
+            const expectedRange = new vscode.Range(
+                new vscode.Position(0, 8),
+                new vscode.Position(0, 8));
+            assert.deepEqual(envResult.range, expectedRange, 'asp-for should be selected');
+            const mStr = envResult.contents[0] as vscode.MarkdownString;
+            assert.ok(mStr.value.includes('InputTagHelper.**For**'), `InputTagHelper.For not included in '${mStr.value}'`);
+        }
+
+        hoverResult = await vscode.commands.executeCommand<vscode.Hover[]>(
+            'vscode.executeHoverProvider',
+            cshtmlDoc.uri,
+            new vscode.Position(0, 19));
+
+        assert.ok(hoverResult, 'Should have a hover result for class');
+        if (hoverResult) {
+            assert.equal(hoverResult.length, 1, 'Something else may be providing hover results');
+
+            const result = hoverResult[0];
+            const expectedRange = new vscode.Range(
+                new vscode.Position(0, 19),
+                new vscode.Position(0, 24));
+            assert.deepEqual(result.range, expectedRange, 'class should be selected');
+            const mStr = result.contents[0] as vscode.MarkdownString;
+            assert.ok(mStr.value.includes('class'), `class not included in ${mStr.value}`);
+        }
+    });
+
+    // MvcWithComponents doesn't find TagHelpers because of test setup foibles.
```

Comment:
Interesting. Does it find it when running it with F5?

Reply:
No. We could probably resolve this if we spent a while on it, but we're able to test it with the other project and to the best of our knowledge it's just a problem with how the test is set up so I figured it wasn't worth it.